plugins/pulp_rpm/plugins/migrations/0036_modulo_render_system.py

```diff
@@ -0,0 +1,38 @@
+"""
+Migration for the new rendering system.
+"""
+
+from pulp.server.db import connection
+
+
+def migrate(*args, **kwargs):
+    """
+    Perform the migration as described in this module's docblock.
+    :param args:   unused
+    :type  args:   list
+    :param kwargs: unused
+    :type  kwargs: dict
+    """
+    db = connection.get_database()
+
+    # Escape all %
+    for u in db.units_rpm.find():
+        u["repodata"]["filelists"] = u["repodata"]["filelists"].replace("%", "%%")
+        u["repodata"]["primary"] = u["repodata"]["primary"].replace("%", "%%")
+        u["repodata"]["other"] = u["repodata"]["other"].replace("%", "%%")
+        db.units_rpm.save(u)
+
+    # Replace old django tags with new modulo ones
+    db.eval("db.units_rpm.find().forEach(function(e,i) {"
+            # filelists - {{ pkgid }}
+            "e.repodata.filelists=e.repodata.filelists.replace("
+            "'{{ pkgid }}','%(pkgid)s');"
+            # primary - {{ checksum }} & {{ checksumtype }}
+            "e.repodata.primary=e.repodata.primary.replace("
+            "'{{ checksum }}','%(checksum)s').replace("
+            "'{{ checksumtype }}','$(checksumtype)s');"
+            # other - {{ pkgid }}
+            "e.repodata.other=e.repodata.other.replace("
+            "'{{ pkgid }}','%(pkgid)s');"
```

Comment:
If someone will add one of the patterns you are trying to substitute into metadata, we will modify the parts of metadata which we are not suppose to touch...
For example, if there will be '{{ pkgid }}' in the changelog, after publish one will find a checksum instead of the expected string '{{ pkgid }}'.